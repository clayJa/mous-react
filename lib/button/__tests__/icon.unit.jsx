import * as renderer from 'react-test-renderer';
import * as React from 'react';
import Button from '../button';
import {mount} from 'enzyme';

describe('icon', () => {
    it('render successfully', () => {
        const json = renderer.create(<Button name="wechat"/>).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('onClick',() => {
        const fn = jest.fn()
        const component = mount(<Button name="wechat" onClick={fn}/>)
        component.find('svg').simulate('click');
        expect(fn).toBeCalled()
    })
});
