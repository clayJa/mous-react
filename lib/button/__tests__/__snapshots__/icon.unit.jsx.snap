// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render successfully 1`] = `
<svg
  className="mous-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
